### DISCUSSION TITLE
Is there any way to handle structures with `const` members?

### DISCUSSION THREAD

**Discussion by lukegv** [nlohmann/json#3921] [Q&A]
Hello! 

I have some data structures that contain `const` members in order to maintain a level of thread safety within the application. 
When writing `from/to_json` functions, I have run into a couple of problems that I am unable to navigate currently. 
Here is what I have:
```
struct DictionarySuggestion
{
    const std::string dictionaryName;
    const std::vector<std::string> suggestions;
};
```
and 
```
void from_json(nlohmann::json &json, DictionarySuggestion&suggestion)
{
    std::vector<std::string> *suggestionPtr = const_cast<std::vector<std::string> *>(&suggestion.suggestions);
    *suggestionPtr = json["suggestions"];
    std::string *dictionaryPtr = const_cast<std::string *>(&suggestion.dictionaryName);
    *dictionaryPtr = json["dictionaryName"];
}

void to_json(const nlohmann::json &json, const DictionarySuggestion&suggestion)
{
    json["dictionaryName"] = suggestion.dictionaryName;
    json["suggestions"] = suggestion.suggestions;
}
```
In the `to_json` method, I am getting an error when I try to assign dictionaryName: 
`C++ no operator matches these operands: operand types are: const nlohmann::json_abi_v3_11_2::basic_json<std::map, std::vector, std::string, bool, int64_t, uint64_t, double, std::allocator, nlohmann::json_abi_v3_11_2::adl_serializer, std::vector<uint8_t, std::allocator<uint8_t>>> = const std::string`
And a similar error for the `suggestion` type also

How can I assign a `const std::string` to the `nlohmann::json` object?

Note: I am aware that const_cast can lead to undefined behaviour, but I am not sure if there is another way to accomplish what we are trying to do, so it may be a necessary evil for the time being. If anyone can suggest another approach, I'd be happy to hear it.

**Comment by gregmarr** [comment]
There is no safe way to write `from_json` for this.  What is it that holds your `DictionarySuggestion` objects?  You may need to do the handling at that level.
